src/librustc/dep_graph/graph.rs

```diff
@@ -76,11 +77,13 @@ impl DepGraph {
         op()
     }
 
-    pub fn with_task<OP,R>(&self, key: DepNode<DefId>, op: OP) -> R
-        where OP: FnOnce() -> R
+    pub fn with_task<C, A, R>(&self, key: DepNode<DefId>, cx: C, arg: A, task: fn(C, A) -> R) -> R
```

Comment:
A comment explaining the setup would be good here, especially why there's a distinction between `cx` and `arg`.